spec/n_plus_one_control/rspec/matchers/perform_exact_number_of_queries_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe NPlusOneControl::RSpec do
+  context "when number of queries match exactly", :n_plus_one do
+    populate { |n| create_list(:post, n) }
+
+    specify do
+      expect { Post.take }
+        .to perform_constant_number_of_queries.exactly(1)
+    end
+  end
+
+  context "when number of queries do not match exactly", :n_plus_one do
+    populate { |n| create_list(:post, n) }
+
+    specify do
+      expect do
+        expect { Post.find_each { |p| p.user.name } }
+          .to perform_constant_number_of_queries.exactly(1)
+      end.to raise_error(RSpec::Expectations::ExpectationNotMetError)
+    end
+  end
+
+  context "exact number has precedence over scale factor", :n_plus_one do
+    specify do
+      expected_params = {population: nil, matching: /INSERT/, scale_factors: [1]}
+      executor = NPlusOneControl::Executor.new(**expected_params)
+      expect(NPlusOneControl::Executor).to receive(:new).with(**expected_params).and_return(executor)
+      expect { Post.take }.to perform_constant_number_of_queries.exactly(0).matching(/INSERT/).with_scale_factors(2, 3)
```

Comment:
For some reason if I use `exactly(1)` with the default `SELECT` matching, in rails 7 tests **two** queries are more often than not detected. When I used pry to debug why, it didn't happen. So there might be some race in newer rails versions. That's why I decided to use `matching(/INSERT/)` as a workaround.

But it is still super strange. Do you have any possible explanation?